**Data Exploration**: Load the dataset into a DataFrame and display the first five rows. What insights can you gather about the structure and types of data present?


In [216]:
import pandas as pd
import numpy as np
import seaborn as sn

In [228]:
# Load the data
data = pd.read_csv("Mobiles_Dataset_2025.csv", sep = ",", encoding="ISO-8859-1")
data.head()

Company Name            Model Name Mobile Weight  RAM Front Camera  \
0        Apple       iPhone 16 128GB          174g  6GB         12MP   
1        Apple       iPhone 16 256GB          174g  6GB         12MP   
2        Apple       iPhone 16 512GB          174g  6GB         12MP   
3        Apple  iPhone 16 Plus 128GB          203g  6GB         12MP   
4        Apple  iPhone 16 Plus 256GB          203g  6GB         12MP   

  Back Camera   Processor Battery Capacity Screen Size  \
0        48MP  A17 Bionic         3,600mAh  6.1 inches   
1        48MP  A17 Bionic         3,600mAh  6.1 inches   
2        48MP  A17 Bionic         3,600mAh  6.1 inches   
3        48MP  A17 Bionic         4,200mAh  6.7 inches   
4        48MP  A17 Bionic         4,200mAh  6.7 inches   

  Launched Price (Pakistan) Launched Price (India) Launched Price (China)  \
0               PKR 224,999             INR 79,999              CNY 5,799   
1               PKR 234,999             INR 84,999              CNY 6,099   
2               PKR 244,999             INR 89,999              CNY 6,499   
3               PKR 249,999             INR 89,999              CNY 6,199   
4               PKR 259,999             INR 94,999              CNY 6,499   

  Launched Price (USA) Launched Price (Dubai)  Launched Year  
0              USD 799              AED 2,799           2024  
1              USD 849              AED 2,999           2024  
2              USD 899              AED 3,199           2024  
3              USD 899              AED 3,199           2024  
4              USD 949              AED 3,399           2024

In [229]:
data.dtypes

Company Name                 object
Model Name                   object
Mobile Weight                object
RAM                          object
Front Camera                 object
Back Camera                  object
Processor                    object
Battery Capacity             object
Screen Size                  object
Launched Price (Pakistan)    object
Launched Price (India)       object
Launched Price (China)       object
Launched Price (USA)         object
Launched Price (Dubai)       object
Launched Year                 int64
dtype: object

**Data Cleaning**: Check for missing values in the dataset. Which columns contain missing values, and how would you handle them?

In [230]:
data.isna().sum()
# No missing data


Company Name                 0
Model Name                   0
Mobile Weight                0
RAM                          0
Front Camera                 0
Back Camera                  0
Processor                    0
Battery Capacity             0
Screen Size                  0
Launched Price (Pakistan)    0
Launched Price (India)       0
Launched Price (China)       0
Launched Price (USA)         0
Launched Price (Dubai)       0
Launched Year                0
dtype: int64

In [231]:
# Remane columns

data.rename(columns={"Launched Price (Pakistan)" : "P_Pakistan", "Launched Price (India)": "P_India", "Launched Price (China)" : "P_China", 
                     "Launched Price (USA)" : "P_USA", "Launched Price (Dubai)" : "P_Dubai"}, inplace=True)                    
data.head()

Company Name            Model Name Mobile Weight  RAM Front Camera  \
0        Apple       iPhone 16 128GB          174g  6GB         12MP   
1        Apple       iPhone 16 256GB          174g  6GB         12MP   
2        Apple       iPhone 16 512GB          174g  6GB         12MP   
3        Apple  iPhone 16 Plus 128GB          203g  6GB         12MP   
4        Apple  iPhone 16 Plus 256GB          203g  6GB         12MP   

  Back Camera   Processor Battery Capacity Screen Size   P_Pakistan  \
0        48MP  A17 Bionic         3,600mAh  6.1 inches  PKR 224,999   
1        48MP  A17 Bionic         3,600mAh  6.1 inches  PKR 234,999   
2        48MP  A17 Bionic         3,600mAh  6.1 inches  PKR 244,999   
3        48MP  A17 Bionic         4,200mAh  6.7 inches  PKR 249,999   
4        48MP  A17 Bionic         4,200mAh  6.7 inches  PKR 259,999   

      P_India    P_China    P_USA    P_Dubai  Launched Year  
0  INR 79,999  CNY 5,799  USD 799  AED 2,799           2024  
1  INR 84,999  CNY 6,099  USD 849  AED 2,999           2024  
2  INR 89,999  CNY 6,499  USD 899  AED 3,199           2024  
3  INR 89,999  CNY 6,199  USD 899  AED 3,199           2024  
4  INR 94,999  CNY 6,499  USD 949  AED 3,399           2024

In [232]:
data["P_USA"] = data["P_USA"].str.replace('USD', '').str.replace(',', '').astype(float)
data["P_India"] = data["P_India"].str.replace('INR', '').str.replace(',', '').astype(float)
data["P_Dubai"] = data["P_Dubai"].str.replace('AED ', '').str.replace(',', '').astype(float)
data.head()

Company Name            Model Name Mobile Weight  RAM Front Camera  \
0        Apple       iPhone 16 128GB          174g  6GB         12MP   
1        Apple       iPhone 16 256GB          174g  6GB         12MP   
2        Apple       iPhone 16 512GB          174g  6GB         12MP   
3        Apple  iPhone 16 Plus 128GB          203g  6GB         12MP   
4        Apple  iPhone 16 Plus 256GB          203g  6GB         12MP   

  Back Camera   Processor Battery Capacity Screen Size   P_Pakistan  P_India  \
0        48MP  A17 Bionic         3,600mAh  6.1 inches  PKR 224,999  79999.0   
1        48MP  A17 Bionic         3,600mAh  6.1 inches  PKR 234,999  84999.0   
2        48MP  A17 Bionic         3,600mAh  6.1 inches  PKR 244,999  89999.0   
3        48MP  A17 Bionic         4,200mAh  6.7 inches  PKR 249,999  89999.0   
4        48MP  A17 Bionic         4,200mAh  6.7 inches  PKR 259,999  94999.0   

     P_China  P_USA  P_Dubai  Launched Year  
0  CNY 5,799  799.0   2799.0           2024  
1  CNY 6,099  849.0   2999.0           2024  
2  CNY 6,499  899.0   3199.0           2024  
3  CNY 6,199  899.0   3199.0           2024  
4  CNY 6,499  949.0   3399.0           2024

In [233]:
#data["P_Pakistan"] = data["P_Pakistan"].str.replace('PKR', '').str.replace(',', '').astype(float)
data["P_Pakistan"] = data["P_Pakistan"].astype(str)

# Remove "AED " and commas, then handle missing values before conversion
data["P_Pakistan"] = data["P_Pakistan"].str.replace("PKR", "", regex=False).str.replace(",", "", regex=False).astype(float)

data.head()

ValueError: could not convert string to float: 'Not available'

In [207]:
data["P_China"] = data["P_China"].str.replace('CNY', '').str.replace(',', '').astype(float)

ValueError: could not convert string to float: '¥13999'

In [162]:

# Remove the currency letters
#data[["P_Pakistan", "P_India", "P_China", "P_USA", "P_Dubai"]] = data[["P_Pakistan", "P_India", "P_China", "P_USA", "P_Dubai"]].replace(r"[a-zA-Z^]", "", regex=True)
data.dtype

Company Name         object
Model Name           object
Mobile Weight        object
RAM                  object
Front Camera         object
Back Camera          object
Processor            object
Battery Capacity     object
Screen Size          object
P_Pakistan           object
P_India              object
P_China              object
P_USA               float64
P_Dubai              object
Launched Year         int64
dtype: object

In [141]:
# Convert to float
data.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 930 entries, 0 to 929
Data columns (total 15 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   Company Name      930 non-null    object
 1   Model Name        930 non-null    object
 2   Mobile Weight     930 non-null    object
 3   RAM               930 non-null    object
 4   Front Camera      930 non-null    object
 5   Back Camera       930 non-null    object
 6   Processor         930 non-null    object
 7   Battery Capacity  930 non-null    object
 8   Screen Size       930 non-null    object
 9   P_Pakistan        930 non-null    object
 10  P_India           930 non-null    object
 11  P_China           930 non-null    object
 12  P_USA             930 non-null    object
 13  P_Dubai           930 non-null    object
 14  Launched Year     930 non-null    int64 
dtypes: int64(1), object(14)
memory usage: 109.1+ KB


In [142]:
data["P_USA"] = data["P_USA"].str.replace('USD', '').str.replace(',', '').astype(float)

In [85]:
# CONVERT the launch prices to USD
data["P_Pakistan"] = data["P_Pakistan"] * 0.0036
data["P_India"] = data["P_India"] * 0.012
data["P_China"] = data["P_China"] * 0.14
data["P_Dubai"] = data["P_Dubai"] * 0.27

TypeError: can't multiply sequence by non-int of type 'float'

**Statistical Analysis**: Calculate the average RAM size and battery capacity of all smartphones in the dataset. What is the mean RAM and battery capacity?


**Filtering Data**: Write a query to filter out all smartphones launched in 2020 with a battery capacity of at least 4000 mAh and a screen size of 6.5 inches or more.

**Comparing Price Trends**: Plot the average launch price of smartphones in India over the years. What trend do you observe?

**Brand Analysis**: Identify the top three brands with the highest average back camera resolution. What are the results?

**Feature Correlation**: Compute the correlation between RAM size and launch price. Does a higher RAM generally correlate with a higher price?

**Processor Analysis**: Find the most commonly used processor among all smartphones. Which processor appears most frequently?

**Price Comparison Across Regions**: Select a smartphone model and compare its launch price across all available countries. Which country had the highest and lowest launch price?

**Trend Prediction**: Based on historical data, predict the launch price of a new smartphone with 8GB RAM, a 108MP back camera, a 5000mAh battery, and a 6.7-inch screen. What would be a reasonable estimate?